commit 78ec789bd397249a9bb412ca91bd360079e7b446
Author: Christian Lamparter <chunkeey@googlemail.com>
Date:   Fri Sep 17 22:58:40 2010 +0200

    carl9170: fix state downgrade during reset
    
    Don't mark the device as completely dead just yet.
    If all goes to plan and carl9170_reboot succeeds
    then we can skip the expensive userspace-driven
    reinitialization anyway.
    
    And if it doesn't and carl9170_reboot fails,
    then carl9170_usb_cancel_urbs will do the
    necessary steps.
    
    Signed-off-by: Christian Lamparter <chunkeey@googlemail.com>
    Signed-off-by: John W. Linville <linville@tuxdriver.com>

--- a/drivers/net/wireless/ath/carl9170/usb.c
+++ b/drivers/net/wireless/ath/carl9170/usb.c
@@ -846,8 +846,6 @@ int carl9170_usb_restart(struct ar9170 *
 
 	carl9170_usb_stop(ar);
 
-	carl9170_set_state(ar, CARL9170_UNKNOWN_STATE);
-
 	if (err)
 		goto err_out;
 
